File: data/src/main/java/org/oppia/data/backends/gae/model/Hint.java
package org.oppia.data.backends.gae.model;

import com.google.auto.value.AutoValue;
import com.google.gson.Gson;
import com.google.gson.TypeAdapter;
import com.google.gson.annotations.SerializedName;

@AutoValue
public abstract class Hint {

  /*
   * Parent Hierarchy: ExplorationContainer -> Exploration -> State -> InteractionInstance -> Hint
   */

  //TODO: AutoJSON null data error

  @SerializedName("hint_content")
  public abstract SubtitledHtml getHintContent();

  public static TypeAdapter<Hint> createTypeAdapter(Gson gson) {
    return new AutoValue_Hint.GsonTypeAdapter(gson);
  }

}

File: data/src/main/java/org/oppia/data/backends/gae/model/Hint.kt
package org.oppia.data.backends.gae.model

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class Hint (

  @Json(name = "hint_content") val hintContent: SubtitledHtml?

)
